### PR TITLE
Changed how import errors are presented as well as ability to change passwords for the current user.

### LINKED ISSUE #1703: Protocol error when creating elasticsearch index
Here is my setup: 
On docker:
timesketch-worker
timesketch-web
nginx (configured to use my own SSL cert)
redis

on DBaaS:
Elasticsearch (Has it's own certificate which has been imported into /etc/ssl/certs)
Postgres (Has it's own certificate which has been imported into /etc/ssl/certs)

So far, setup is fine. Able to create a sketch, and moved to other pages. Previously I had SSL errors connecting to ES, but that has been fixed now. 

Here's my timesketch.conf
--------------------------------
ELASTIC_HOST = 'xxxx.private.databases.appdomain.cloud'
ELASTIC_PORT = 'xxx'
ELASTIC_USER = 'admin'
ELASTIC_PASSWORD = 'xxxx'
ELASTIC_SSL = '/etc/ssl/certs/c92a5016-d56a-11e9-baa1-56da0ec946bd'
ELASTIC_VERIFY_CERTS = True
----------------------

Problem comes when I tried to upload via web or importer. 
root@us-vpc-dallas-timesketch-uploader:~# timesketch_importer xxxx
[2021-03-21 16:10:03,157] timesketch_importer.importer_frontend/INFO Creating a client.
[2021-03-21 16:10:03,689] timesketch_importer.importer_frontend/INFO Client created.
[2021-03-21 16:10:03,689] timesketch_importer.importer_frontend/INFO Saving TS config.
[2021-03-21 16:10:03,690] timesketch_importer.importer_frontend/INFO Uploading file.
[2021-03-21 16:10:03,694] timesketch_importer.importer_frontend/INFO About to upload file.
[2021-03-21 16:10:27,871] timesketch_importer.importer_frontend/INFO File upload completed.
Checking file upload status: ..[FAIL]
Unable to index timeline, reason: 2021-03-21 16:10:29,732 [INFO] (MainProcess) PID:24 <data_location> Determined data location: /usr/share/plaso
2021-03-21 16:10:30,249 [INFO] (MainProcess) PID:24 <elastic_ts> Timeline identifier: 11
**Unable to create Elasticsearch index with error: ConnectionError(('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))) caused by: ProtocolError(('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer')))**

Which app is creating the ES Index? This is because of ELASTIC_SSL error. I think it is not being passed to the app trying to create the ES index.

### LINKED ISSUE #1708: Export to CSV not working as expected
**Describe the bug**
Export to CSV doesn't take into account selected timelinename when exporting to CSV.

**To Reproduce**
1. Import multiple timelines in a sketch.
2. choose a single timeline. 
3. do a generic search such as "FILE_NAME".
4. hit export to CSV

exported data content will be the search results for "FILE_NAME" but on all imported timelines.

**Expected behavior**
Data exported to CSV should be exactly the same as the data displayed based on the selected search parameters  

**Screenshots**
N/A

**Desktop (please complete the following information):**
-OS:Ubuntu 20.04.2 LTS
-Browser : Firefox
-Version: 86.0

**Additional context**
-Timesketch version: 20210224

### PR DESCRIPTION
**Pull request by kiddinn**
This PR provides few new functionalities:

+ The ability to change the password for the current user in the API
+ The ability to change the password inside the API client.
+ Minor change in the sketch API client object to make it a bit more error tolerant.
+ Changing the release label of the docker config to use Ubuntu 20.04
+ Adding a new entry into the DataSource model, `error_message`
+ Changing the import errors of a timeline to the DataSource model instead of changing the timeline description field.
+ Updating versions.
+ Data upload: When data is uploaded with the same name as a previous timeline, but for some reasons a different index is chosen, current behavior is to fail/abort the upload, changed the behavior to instead change the timeline name and retry the upload.
+ Data upload: Adds in the option to psort to provide user/pass information to Elastic.
+ Timeline export: Fixed an issue with timeline exports not honoring timeline IDs.

**Checks**

- [x] All tests succeed.
- [x] Unit tests added.
- [x] e2e tests added.
- [x] Documentation updated.

**Closing issues**

+ This PR does implement the API and API client aspect of #605
+ closes #1703 
+ closes #1708

**Missing UI Elements**

There are two things that need to be changed in the UI as a result of this PR:

1. Change how errors in timelines are presented, from the `timeline.description` to the `data_source.error_message`
2. Add the ability for a user to change their own password (see #605)

### CODE REVIEW

**Comment by berggren** [comment]
UI related issue: #1707

File: api_client/python/timesketch_api_client/user.py [hunk 1]
```diff
@@ -43,6 +45,38 @@ def _get_data(self):
 
         return self._object_data
 
+    def change_password(self, new_password=''):
```
Comment (berggren) [inline api_client/python/timesketch_api_client/user.py:48]: Do we really want CLI user interface in the API client? Isn't this the responsibility of the user frontend client (e.g. Colab magics, picatrix, the CLI client)?

File: timesketch/api/v1/resources/upload.py [hunk 2]
```diff
@@ -162,11 +163,24 @@ def _upload_and_index(
...
-                'but is stored in a different index, check the data_label '
-                'on the uploaded data')
+                'but is stored in a different index: name {0:s} attempting '
+                'index: {1:s} but found index {2:s} - retrying with a '
+                'different timeline name.'.format(
+                    timeline_name, searchindex.index_name,
+                    timeline_.searchindex.index_name))
+
+            timeline_name = '{0:s}_{1:s}'.format(
+                timeline_name, uuid.uuid4().hex)
```
Comment (berggren) [inline timesketch/api/v1/resources/upload.py:175]: This will become a very long name in the UI. Could we generate a shorter uniq string here?

**Review by berggren — CHANGES_REQUESTED** [review]

File: timesketch/api/v1/resources/upload.py (hunk 2 above)
Comment (kiddinn) [reply timesketch/api/v1/resources/upload.py:175]: forgot to shorten this, the idea was to just use the last five characters.

File: api_client/python/timesketch_api_client/user.py (hunk 1 above)
Comment (kiddinn) [reply api_client/python/timesketch_api_client/user.py]: sure, let's remove the `cli_input` logic

**Review by kiddinn — COMMENTED** [review]
PTAL

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**